File: app/src/main/java/com/canli/oya/traininventoryroom/viewmodel/ChosenTrainViewModel.java
package com.canli.oya.traininventoryroom.viewmodel;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import com.canli.oya.traininventoryroom.data.TrainDatabase;
import com.canli.oya.traininventoryroom.data.entities.TrainEntry;

public class ChosenTrainViewModel extends ViewModel {
    private final LiveData<TrainEntry> chosenTrain;

    ChosenTrainViewModel(TrainDatabase database, int trainId) {
        chosenTrain = database.trainDao().getChosenTrain(trainId);
    }

    public LiveData<TrainEntry> getChosenTrain() {
        return chosenTrain;
    }

}

File: app/src/main/java/com/canli/oya/traininventoryroom/viewmodel/ChosenTrainViewModel.kt
package com.canli.oya.traininventoryroom.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel

import com.canli.oya.traininventoryroom.data.TrainDatabase
import com.canli.oya.traininventoryroom.data.TrainEntry

class ChosenTrainViewModel internal constructor(database: TrainDatabase, trainId: Int) : ViewModel() {
    val chosenTrain: LiveData<TrainEntry> = database.trainDao().getChosenTrain(trainId)
}
